frontend/src/app/auth.service.ts

```diff
@@ -33,16 +33,22 @@ export class AuthService {
     private http: HttpClient,
     private router: Router,
   ) {
-    this.http.get<User>(this.currentUserUrl)
+    this.getUser()
       .subscribe(user => {
-        if (user.username) {
+        if (user) {
           this.changeUserEvent.next(deepFreeze(user));
         } else {
           this.changeUserEvent.next(null);
         }
       });
   }
 
+  getUser(): Observable<User | null> {
+    return this.http.get<User>(this.currentUserUrl).pipe(
+      map(user => user.username ? user : null),
```

Comment:
Adding to my previous question, I am not neccessecarily keen on this map statement - if we really want to get 'null' from the request, maybe we could change the backend accordingly?

Reply:
@Magul said: null isn't correctly returned as JSON, so are you sure you want backend to return null?

Reply:
Well that's a bummer, since 'null' is a valid JSON value IMO perfect for this situation. I guess we should live with it then.

Reply:
My mistake, we were trying to send empty string from backend and it ends with
```
JSON.parse('')
VM249:1 Uncaught SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at <anonymous>:1:6
```
but `null` is working just fine.

if so I'm :+1: with changing how backend in that manner works.